src/main/java/org/broadinstitute/hellbender/utils/test/SamAssertionUtils.java

```diff
@@ -135,4 +109,126 @@ public static void assertSamValid(final File sam) throws IOException {
         assertSamValid(sam, ValidationStringency.LENIENT, null);
     }
 
+    /**
+     * Compares SAM/BAM files in a stringent way but not by byte identity (allow reorder of attributes)
+     */
+    private static String samsEqualLenient(File sam1, File sam2, ValidationStringency validation, File reference) throws IOException {
```

Comment:
Document your confusing scheme of returning a null `String` from this method when the files are equal.